File: app/src/main/groovy/us/kesslern/ascient/service/AndroidIdService.java
package us.kesslern.ascient.service;

import android.app.Activity;
import android.provider.Settings;

import static android.provider.Settings.*;

public class AndroidIdService {
    public static String getAndroidId(Activity activity) {
        return Secure.getString(activity
                        .getApplicationContext()
                        .getContentResolver(),
                        Secure.ANDROID_ID);
    }
}

File: app/src/main/java/us/kesslern/ascient/service/AndroidIdService.kt
package us.kesslern.ascient.service

import android.app.Activity
import android.provider.Settings

import android.provider.Settings.*

object AndroidIdService {
    fun getAndroidId(activity: Activity): String {
        return Secure.getString(
                activity.applicationContext
                        .contentResolver,
                Secure.ANDROID_ID)
    }
}
